src/docs2vecs/subcommands/indexer/skills/ada002_embedding_skill.py

```diff
@@ -11,14 +11,23 @@ class AzureAda002EmbeddingSkill(IndexerSkill):
     def __init__(self, config: dict, global_config: Config):
         super().__init__(config, global_config)
 
-    def az_ada002_embeddings(self, content: str):
+    def az_ada002_embeddings(self, content: str, chunk_id=None):
+        self.logger.debug(
+            f"Requesting embedding for chunk_id={chunk_id}, content_length={len(content)}"
+        )
         embed_model = AzureOpenAIEmbedding(
             deployment_name=self._config["deployment_name"],
             api_key=self._config["api_key"],
             azure_endpoint=self._config["endpoint"],
             api_version=self._config["api_version"],
         )
-        return embed_model.get_query_embedding(content)
+        try:
+            embedding = embed_model.get_query_embedding(content)
+            self.logger.debug(f"Received embedding for chunk_id={chunk_id}")
+            return embedding
+        except Exception as e:
+            self.logger.error(f"Embedding failed for chunk_id={chunk_id}: {e}")
+            return None
```

Comment:
This is changing the strategy of the error handling. For better or worse, not sure but that's a significant change indeed. 
Instead of letting the exception to freely bubble up it is intercepted and converted into None.

@dpomian or @JayashriGovindan : what do you think about that? Does it play well with the rest of the code?